src/config.rs

```diff
@@ -23,7 +28,9 @@ impl ConfigBuilderExt for ConfigBuilder<ClientConfig, WantsVerifier> {
     #[cfg(feature = "rustls-native-certs")]
     #[cfg_attr(docsrs, doc(cfg(feature = "rustls-native-certs")))]
     #[cfg_attr(not(feature = "logging"), allow(unused_variables))]
-    fn with_native_roots(self) -> ConfigBuilder<ClientConfig, WantsTransparencyPolicyOrClientCert> {
+    fn with_native_roots(
+        self,
+    ) -> Option<ConfigBuilder<ClientConfig, WantsTransparencyPolicyOrClientCert>> {
```

Comment:
my feeling is that this should be returning a `Result` with a `Box<dyn std::error::Error>` or `std::io::Error`

then the `.expect("could not load platform certs")` line below can become `?` to avoid a reachable panic there.